README.md

```diff
@@ -11,11 +27,24 @@ Role Variables
 * `postgres_locale`: Locale used for postgres databases (default to `en_US.UTF-8`).
 * `postgres_default_roles`: List of default user roles (default to `NOSUPERUSER,NOCREATEDB,NOCREATEROLE`).
 
+* `postgres_listen_address`: Listening address of Postgres (default to `localhost`).
+* `postgres_listen_port`: Listening port of Postgres (default to `5432`).
+
 * `postgres_users`: Dict of users and databases to creates with username as key.
     * `password`: User password.
     * `roles`: User roles (use `postgres_default_roles` if not set).
     * `database`: Name of database to create (default to the user name).
 
+* `postgres_clients`: List of clients allowed to connect.
+    * `db`: Required.
+    * `user`: Required.
+    * `address`: Default to `127.0.0.1/32`.
+    * `peer`: Default to `md5`.
+    * `type`: Default to `host`.
+
+* `postgres_conf_template`: Template for generating `postgresql.conf` (default to `etc/postgresql/10/postgresql.conf`).
```

Comment:
Is the value here that we don't have to list every possible config option? It seems like if someone used a different template the options from above (postgres_listen_port and postgres_clients) would be moot?

Reply:
Exactly. I think allowing a bit of customization but not too much is the way to go. In the case you need heavy custom it's easier to provide a different templates and not dozens of variables.
There is the same feature in pyslackers/ansible-role-nginx. 

Yeah the variables won't be used except if the new templates uses them.

Reply:
I guess I don't see the value there much and I, speaking as a user, would rather have a clear API (e.g. config params) that does the file/template for me (vs me, as a user of the role, needing to know how to set up a base template + know how the postgres file looks).

Reply:
Well the goal is to have both. I don't want to overwhelm the users with too much config params but on the other hand there is value is using this role without our configuration.

I think documenting them in `defaults/main.yml` and not in the readme would give us the best of both world. What do you think of that ?

Reply:
Seems reasonable, you are one of the primary users outside the limited pyslackers cases - so I lean on your needs a bit :)